src/GitHub.VisualStudio/UI/GitHubPane.cs

```diff
@@ -75,25 +83,30 @@ public GitHubPane() : base(null)
 
         protected override void Initialize()
         {
-            base.Initialize();
-            Initialize(this);
+            InitializeAsync().Catch(ShowError).Forget();
         }
 
-        public void Initialize(IServiceProvider serviceProvider)
+        public Task<IGitHubPaneViewModel> GetViewModelAsync() => viewModelSource.Task;
```

Comment:
This looks like it will likely be misused to violate the [3rd threading rule](https://github.com/Microsoft/vs-threading/blob/master/doc/threading_rules.md#rule-3-if-ever-awaiting-work-that-was-started-earlier-that-work-must-be-joined). If someone tries to await this Task within the context of a JTF.Run delegate, you'll quite likely deadlock.
Can you replace your TaskCompletionSource with a JoinableTask instead? Then call `InitializeAsync` within a JTF.RunAsync delegate and store the resulting `JoinableTask` in your field.
Then *this* method should return `joinableTask.JoinAsync()` (or simply `await joinableTask`) to mitigate deadlocks in callers.